pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -180,7 +180,6 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo) (*apps.Dep
 	defaultArgs := []string{
 		"--foreground",
 		"--id", osdID,
-		"--conf", osd.Config,
```

Comment:
If we don't ever need the --conf option, we should be able to remove the `config-init` container that generates the config file. I do wonder if the filestore osds still need this, but maybe it'll be fine since this change is just for the daemon pod and does not affect the prepare pod. In any case, the integration tests should show if it's still needed.

Reply:
The config-init container is needed to bootstrap the osd, so the info from the config is important. But since ceph-volume is really the default for everything and it can't be changed, we might consider removing that file/struct/pod as a whole.

Reply:
Ok, looks like the `defaultArgs` were only used in two scenarios:
- A filestore OSD configured on a device (not a common scenario, especially in production)
- A bluestore OSD on a device that was not configured with ceph-volume. This is likely the case where users were not seeing their overrides applied.

I see now why this change would have been missed in our testing.